src/sagemaker/model.py

```diff
@@ -364,12 +367,49 @@ def register(
         model_package = self.sagemaker_session.create_model_package_from_containers(
             **model_pkg_args
         )
+        if isinstance(self.sagemaker_session, PipelineSession):
+            return None
         return ModelPackage(
             role=self.role,
             model_data=self.model_data,
             model_package_arn=model_package.get("ModelPackageArn"),
         )
 
+    @runnable_by_pipeline
+    def create(
```

Comment:
Thanks for reviewing! 
I also thought about this but considering the backward compatibility, I decided to wrap a `create` method on top of `_create_sagemaker_model`.
Here're my concerns:
- `_create_sagemaker_model` has multiple usages inside the code base (6 in src and >12 in tests). Replacing it can add much more code changes to the already extremely large PR and I'm afraid it may bring some risks to this important and new feature.
Thus, I'm thinking we can add a TODO note here and open a separate PR to replace it.

Reply:
Thanks, I understand if we decide to keep `_create_sagemaker_model ` as well for backwards compatibility, ideally consumers however shouldn't use it directly and we can add a Todo for this.